src/lib/insights/listener.tsx

```diff
@@ -63,15 +63,17 @@ const insightsListener = (BaseComponent: any) => {
         }
       }
 
-      // old way, e.g. instantsearch.insights("clickedObjectIDsAfterSearch", { .. })
-      const insightsTarget = findInsightsTarget(
-        event.target as HTMLElement | null,
-        event.currentTarget as HTMLElement | null,
-        element => hasDataAttributes(element)
-      );
-      if (insightsTarget) {
-        const { method, payload } = readDataAttributes(insightsTarget);
-        props.insights(method, payload);
+      if (__KEEP_DEPRECATION__) {
```

Comment:
What do you see as the advantages of this vs. a comment // @MAJOR? 

I see how we can then expose a build without deprecated code, but will that difference be significant enough to justify the change?

Reply:
In my opinion, it will enable us to test(unit, e2e) with different builds.

By the way, I'm not suggesting to change the current ones we already wrote. It will be too much.

Reply:
I wonder actually if it might not be better to wrap this code in a conditional similar to `warn`, but specifically for deprecated code. That way we force deprecated code that's still called to have a warning, and could strip out all deprecated code in a similar fashion as code after a warning

Reply:
that's a nicer way

Reply:
> What do you see as the advantages of this vs. a comment // @major?

Thank you for tagging me, but I'm not sure I can be much help here. I wish you the best, though! 🤗